controller/lib/hal/hal.h

```diff
@@ -207,7 +214,16 @@ class HalApi {
   void test_serialPutIncomingData(const char *data, uint16_t len);
 #endif
 
+  // Performs the device soft-reset
```

Comment:
Please add comments as to what this does on native (i.e., for now, all the watchdog functions are no-op + possibly a TODO to change this)

Reply:
Same, maybe this should be a HAL_MOCK method so we can test it from gtest?